### PR TITLE
Catch catastrophic solver failure when building MIS

### PR DESCRIPTION
**Pull request by bknueven**
## Summary/Motivation:
The minimal intractable system calculation had a try/except around a solve we expect to fail, but when we check for failure we failed to first check the results object against `None`.

## Changes proposed in this PR:
- Insert `results is not None` conditional to short-circuit, thus not calling `pyo.check_optimal_termination(results)` if `results is None`.

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by mrmundt — APPROVED** [review]

File: pyomo/contrib/iis/mis.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def _constraint_generator():
     except:
         results = None
 
-    if pyo.check_optimal_termination(results):
+    if (results is not None) and pyo.check_optimal_termination(results):
```
Comment (jsiirola) [inline pyomo/contrib/iis/mis.py:290]: Question: would it better to not swallow all exceptions from the `solve()` above and instead use `load_solutions=False` in the call?  Then results should never be `None`?

Comment (bknueven) [reply pyomo/contrib/iis/mis.py:290]: Probably. Would that still not raise an exception if the solver returns a non-zero exit code? I think that's why I've implemented these as blanket `try / except` blocks.

Comment (jsiirola) [reply pyomo/contrib/iis/mis.py:290]: OK.  I would just be worried about the bare `except:` -- that could be suppressing any number of errors and there would be no way to know what was going on.

**Review by jsiirola — APPROVED** [review]
I am OK merging this as it resolves a serious error blocking MIS users.  However, this highlights fragility in the MIS implementation due to the use of a bare (catch-all) "except:" clause.  

It would be good to revisit this and see if the bare "except:" can be removed.

**Pull request merged**